build.gradle

```diff
@@ -469,84 +470,78 @@ task fakePublish {
-                def javadocJar = "${x64Base}/AmazonCorrettoCryptoProvider-javadoc.jar"
-                artifact source: file(x86Jar),classifier:"linux-x86_64",extension:"jar"
-                artifact source: file(aarch64),classifier:"linux-aarch_64",extension:"jar"
-                artifact emptyJar
-                artifact source: file(javadocJar),classifier:"javadoc",extension:"jar"
-                artifact source: file(sources),classifier:"sources",extension:"jar"
-
-                pom {
-                    name = pomName
-                    description = 'The Amazon Corretto Crypto Provider (ACCP) is a collection of high-performance cryptographic implementations exposed via the standard JCA/JCE interfaces.'
-                    url = 'https://github.com/corretto/amazon-corretto-crypto-provider'
-                    licenses {
-                        license {
-                            name = 'The Apache License, Version 2.0'
-                            url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
-                        }
-                    }
-                    scm {
-                        url = 'https://github.com/corretto/amazon-corretto-crypto-provider.git'
-                        connection = 'scm:git:git@github.com:corretto/amazon-corretto-crypto-provider.git'
-                        developerConnection = 'scm:git:git@github.com:corretto/amazon-corretto-crypto-provider.git'
-                    }
-                    developers {
-                        developer {
-                            id = 'amazonwebservices'
-                            organization = 'Amazon Web Services'
-                            organizationUrl = 'https://aws.amazon.com'
-                        }
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
+            def jarBaseName = "AmazonCorrettoCryptoProvider.jar"
+            def x64Base = "${System.env.CODEBUILD_SRC_DIR_Linux_x64_Artifacts}"
+            def aarch64Base = "${System.env.CODEBUILD_SRC_DIR_Linux_aarch64_Artifacts}"
+            def pomName = "Amazon Corretto Crypto Provider"
+            if (isFips) {
+                artifactId = project.name + "-fips"
+                pomName = pomName + " (FIPS)"
+                x64Base = "${System.env.CODEBUILD_SRC_DIR_Linux_x64_Artifacts_FIPS}"
+                aarch64Base = "${System.env.CODEBUILD_SRC_DIR_Linux_aarch64_Artifacts_FIPS}"
+            } else {
+                artifactId = project.name
+            }
+            def x86Jar = "${x64Base}/${jarBaseName}"
+            def aarch64Jar = "${aarch64Base}/${jarBaseName}"
+            def sourceJar = "${x64Base}/AmazonCorrettoCryptoProvider-sources.jar"
+            def javadocJar = "${x64Base}/AmazonCorrettoCryptoProvider-javadoc.jar"
+            artifact source: file(x86Jar),classifier:"linux-x86_64",extension:"jar"
+            artifact source: file(aarch64Jar),classifier:"linux-aarch_64",extension:"jar"
+            artifact emptyJar
```

Comment:
what's the purpose of this?

Reply:
A Maven package *needs* a primary jar, a javadoc jar, and a source jar.

However, we don't have a working primary jar (that would be the not-yet created "uber jar" most likely) so to fill that need we put in an empty jar which does absolutely nothing.

The other two jars (not javadoc, not empty) are the variants for our platforms.

Reply:
ah, makes sense. thanks.